src/eos-google-chrome

```diff
@@ -48,26 +52,68 @@ class GoogleChromeLauncher:
-        chrome_launcher = self._get_chrome_flatpak_launcher()
-        if chrome_launcher:
-            logging.info("Flatpak launcher for Chrome found. Launching...")
-            self._run_chrome_app(chrome_launcher, self._params)
-        else:
+    def check(self):
+        if not self._chrome_launcher:
             logging.info("Could not find flatpak launcher for Chrome. Opening App Center...")
             self._run_app_center_for_chrome()
+            return False
+
+        logging.info("Flatpak launcher for Chrome found.")
+
+        return True
+
+    def start(self):
+        logging.info("Launching...")
+        self._run_chrome_app(self._chrome_launcher, self._params)
 
     def _run_chrome_app(self, chrome_launcher, params):
-            try:
-                launcher_process = subprocess.Popen([chrome_launcher] + params)
-                logging.info("Running Google Chrome launcher with PID %d", launcher_process.pid)
-            except OSError as e:
-                exit_with_error("Could not launch Google Chrome: {}".format(repr(e)))
+        try:
+            launcher_process = subprocess.Popen([chrome_launcher] + params)
+            logging.info("Running Google Chrome launcher with PID %d", launcher_process.pid)
+        except OSError as e:
+            exit_with_error("Could not launch Google Chrome: {}".format(repr(e)))
+
+        launcher_process.wait()
+        logging.info("Google Chrome launcher stopped")
+
+    def _wait_for_installation(self):
+        loop = GLib.MainLoop()
+
+        def handler(*signals):
+            for signal in signals[0]:
+                if signal == dbus.String("google-chrome.desktop"):
+                    loop.quit()
+                    return
+            exit_with_error("Gnome Software installed some new application but not Google Chrome. "
+                            "Most probably, user decided to install something else. Quitting...")
+
+        dbus_glib.DBusGMainLoop(set_as_default=True)
+        bus = dbus.SessionBus()
+        bus.add_signal_receiver(handler, dbus_interface="org.gnome.Shell.AppStore",
+                                signal_name="ApplicationsChanged")
+
+        loop.run()
 
-            launcher_process.wait()
-            logging.info("Google Chrome launcher stopped")
+    def _set_as_default_browser(self):
+        mimetypes = ["text/html",
+                     "x-scheme-handler/http",
+                     "x-scheme-handler/https",
+                     "x-scheme-handler/about"]
+        for mimetype in mimetypes:
+            subprocess.Popen(["xdg-mime", "default", "google-chrome.desktop", mimetype])
+
+        subprocess.Popen(["xdg-settings", "set", "default-web-browser", "google-chrome.desktop"])
+
+    def _touch_done_file(self):
+        chrome_setup_done_path = pathlib.Path(os.path.expanduser("~/.config/google-chrome-initial-setup-done"))
+        if not chrome_setup_done_path.exists():
+            chrome_setup_done_path.touch()
```

Comment:
It shouldn't happen but this can raise a PermissionError exception (and probably others, python doc is not very clear on that). I think we should catch it and print some error message in case there's a problem.